### PR TITLE
Show number of applied/reverted migrations in case of error

### PR DESCRIPTION
**Pull request by Vovan-VE**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14169
| Obsoleted by | PR #15095

### CODE REVIEW

File: framework/console/controllers/BaseMigrateController.php [hunk 1]
```diff
@@ -179,14 +179,22 @@ public function actionUp($limit = 0)
...
         $applied = 0;
         if ($this->confirm('Apply the above ' . ($n === 1 ? 'migration' : 'migrations') . '?')) {
-            foreach ($migrations as $migration) {
-                if (!$this->migrateUp($migration)) {
-                    $this->stdout("\n$applied from $n " . ($applied === 1 ? 'migration was' : 'migrations were') ." applied.\n", Console::FG_RED);
-                    $this->stdout("\nMigration failed. The rest of the migrations are canceled.\n", Console::FG_RED);
+            try {
+                foreach ($migrations as $migration) {
+                    if (!$this->migrateUp($migration)) {
+                        $this->stdout("\n$applied from $n " . ($applied === 1 ? 'migration was' : 'migrations were') ." applied.\n", Console::FG_RED);
```
Comment (samdark) [inline framework/console/controllers/BaseMigrateController.php]: Shouldn't it be replaced with `showSummaryOnError()` call?

File: framework/console/controllers/BaseMigrateController.php [hunk 2]
```diff
@@ -240,14 +248,22 @@ public function actionDown($limit = 1)
...
         $reverted = 0;
         if ($this->confirm('Revert the above ' . ($n === 1 ? 'migration' : 'migrations') . '?')) {
-            foreach ($migrations as $migration) {
-                if (!$this->migrateDown($migration)) {
-                    $this->stdout("\n$reverted from $n " . ($reverted === 1 ? 'migration was' : 'migrations were') ." reverted.\n", Console::FG_RED);
-                    $this->stdout("\nMigration failed. The rest of the migrations are canceled.\n", Console::FG_RED);
+            try {
+                foreach ($migrations as $migration) {
+                    if (!$this->migrateDown($migration)) {
+                        $this->stdout("\n$reverted from $n " . ($reverted === 1 ? 'migration was' : 'migrations were') ." reverted.\n", Console::FG_RED);
```
Comment (samdark) [inline framework/console/controllers/BaseMigrateController.php]: Shouldn't it be replaced with `showSummaryOnError()` call?

File: framework/console/controllers/BaseMigrateController.php [hunk 3]
```diff
@@ -301,19 +317,40 @@ public function actionRedo($limit = 1)
...
 
         if ($this->confirm('Redo the above ' . ($n === 1 ? 'migration' : 'migrations') . '?')) {
-            foreach ($migrations as $migration) {
-                if (!$this->migrateDown($migration)) {
-                    $this->stdout("\nMigration failed. The rest of the migrations are canceled.\n", Console::FG_RED);
+            $reverted = 0;
+            try {
+                foreach ($migrations as $migration) {
+                    if (!$this->migrateDown($migration)) {
+                        $this->stdout("\nMigration failed. The rest of the migrations are canceled.\n", Console::FG_RED);
```
Comment (samdark) [inline framework/console/controllers/BaseMigrateController.php]: Shouldn't it be replaced with `showSummaryOnError()` call?

File: framework/console/controllers/BaseMigrateController.php [hunk 4]
```diff
@@ -301,19 +317,40 @@ public function actionRedo($limit = 1)
...
-            foreach (array_reverse($migrations) as $migration) {
-                if (!$this->migrateUp($migration)) {
-                    $this->stdout("\nMigration failed. The rest of the migrations are canceled.\n", Console::FG_RED);
 
-                    return self::EXIT_CODE_ERROR;
+            $applied = 0;
+            try {
+                foreach (array_reverse($migrations) as $migration) {
+                    if (!$this->migrateUp($migration)) {
+                        $this->stdout("\nMigration failed. The rest of the migrations are canceled.\n", Console::FG_RED);
```
Comment (samdark) [inline framework/console/controllers/BaseMigrateController.php]: Shouldn't it be replaced with `showSummaryOnError()` call?

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/console/controllers/BaseMigrateController.php (hunk 1 above)
Comment (Vovan-VE) [reply framework/console/controllers/BaseMigrateController.php]: Do you mean to extract full `try` to method with which arguments?

**Comment by Vovan-VE** [comment]
This PR in outdated. Now it would be much better to reimplement it in a new PR. What about handling exceptions once inside `migrateUp()` and `migrateDown()` methods to let them just to return `false`? To do it these methods need to show caught exception. Is there another way but `ErrorHandler::convertExceptionToString()` to do it?

**Pull request closed without merging**